app/controllers/katello/api/v2/products_controller.rb

```diff
@@ -10,6 +10,7 @@ class Api::V2::ProductsController < Api::V2::ApiController
     before_action :authorize_ssl_ca_cert, :only => [:update, :create]
     before_action :authorize_ssl_client_cert, :only => [:update, :create]
     before_action :authorize_ssl_client_key, :only => [:update, :create]
+    before_action :find_products, only: [:auto_complete_name]
```

Comment:
Can you check if this is required. I see auto_complete_name is running it's own query.